README.md

```diff
@@ -0,0 +1,241 @@
+<a href="https://packagist.org/packages/nothingworks/blade-svg"><img src="https://poser.pugx.org/nothingworks/blade-svg/license.svg" alt="License"></a>
+
+A package to easily make use of SVG icons in your Laravel Blade views.
+
+- [Icon Packages](#icon-packages)
+- [Requirements](#requirements)
+- [Installation](#installation)
+- [Updating](#updating)
+- [Configuration](#configuration)
+    - [Defining Sets](#defining-sets)
+    - [Icon Paths](#icon-paths)
+    - [Prefixing Icons](#prefixing-icons)
+    - [Default Classes](#default-classes)
+- [Usage](#usage)
+    - [Components](#components)
+    - [Directives](#directives)
+    - [Helper](#helper)
+- [Building Packages](#building-packages)
+- [Changelog](#changelog)
+- [Maintainers](#maintainers)
+- [License](#license)
+
+## Icon Packages
+
+This package is a base package to make it easy for you to use SVG icons in your app. If you want to start using a specific icon set, we offer the following ones below:
+
+- Coming soon...
+
+Feel free to open up an issue on the repo to [request another set](https://github.com/nothingworks/blade-svg/issues/new).
+
+## Requirements
+
+- PHP 7.2 or higher
+- Laravel 7.6 or higher
+
+## Installation
+
+```bash
+composer require nothingworks/blade-svg
+```
+
+After installing the package, publish the configuration and **uncomment** the `default` icon set: 
+
+```bash
+php artisan vendor:publish --tag=blade-icons
+```
+
+## Updating
+
+Please refer to [`the upgrade guide`](UPGRADE.md) when updating the library.
+
+## Configuration
+
+### Defining Sets
+
+Blade Icons support multiple sets. You can define these by passing a key/value combination in the `blade-icons.php` config file's `sets` setting:
+
+```php
+<?php
+
+return [
+    'sets' => [
+        'default' => [
+            'path' => 'resources/svg',
+        ],
+    ],
+];
+```
+
+Feel free to add as many sets as you wish. Blade Icons ships with a `default` set for your app which you may adjust to your liking.
+
+### Icon Paths
+
+If you wanted to add icons from a different directory in your app, say `resources/images/svg`, you can set it like this:
+
+```php
+<?php
+
+return [
+    'sets' => [
+        'default' => [
+            'path' => 'resources/images/svg',
+        ],
+    ],
+];
+```
+
+> Always make sure you're pointing to existing directories.
+
+### Prefixing Icons
+
+In the default icon set the `icon` prefix will be applied to every icon, but you're free to adjust this in the `blade-icons.php` config file:
+
+```php
+<?php
+
+return [
+    'sets' => [
+        'default' => [
+            'prefix' => 'icon',
+        ],
+    ],
+];
+```
+
+Defining a prefix for every set is required and every prefix should be unique.
+
+When referencing icons with the [Blade directive](#directives) or [helper](#helper) you can omit the prefix to reference icons from the `default` set.
```

Comment:
I think it should be worth noting that icons named in kebab-case MUST include the prefix.

For example, for an icon `/resources/svg/arrow-left-circle.svg`

```blade
{{-- This wont work. An error 'Svg by name "right" from set "default" not found' will be thrown--}}
@svg('arrow-left-circle','h-5 text-gray-700')

{{-- This will work --}}
@svg('icon-arrow-left-circle','h-5 text-gray-700')
```

The alternative is to ensure that all icons ARE NOT kebab-cased, instead use camelCase or PascalCase or snake_case. This maybe challenging considering most icon set use kebab-case naming.

Reply:
Ow! You've actually discovered a bug. I fixed this. Default icons don't need the prefix anymore. Thanks for testing!

Reply:
You are welcome.